src/main/java/io/kestra/plugin/notifications/sentry/SentryAlert.java

```diff
@@ -117,29 +125,88 @@ public class SentryAlert extends Task implements RunnableTask<VoidOutput> {
     public VoidOutput run(RunContext runContext) throws Exception {
         String dsn = runContext.render(this.dsn);
 
-        String url = dsn;
+        String storeUrl = dsn;
+        String envelopeUrl = dsn;
         if (dsn.matches(SENTRY_DSN_REGEXP)) {
             String protocol = dsn.split("://")[0];
             String publicKey = dsn.split("@")[0].replace(protocol + "://", "");
             String host = dsn.split("@")[1].split("/")[0];
             String projectId = dsn.split("@")[1].split("/")[1];
             /*
-            To make passing the correct API endpoint URL easier, 
+            To make passing the correct API endpoint URL easier,
             users only need to provide the Sentry DSN, and we parse the required attributes for the URL
-            using the format https://{HOST}/api/{PROJECT_ID}/store/?sentry_version=7&sentry_client=java&sentry_key={PUBLIC_KEY}
+            using the following formats:
+            STORE_URL: https://{HOST}/api/{PROJECT_ID}/store/?sentry_version=7&sentry_client=java&sentry_key={PUBLIC_KEY}
+            ENVELOP_URL: https://{HOST}/api/{PROJECT_ID}/envelope/?sentry_version=7&sentry_client=java&sentry_key={PUBLIC_KEY}
             */
-            url = "%s://%s/api/%s/store/?sentry_version=%s&sentry_client=%s&sentry_key=%s" 
-                .formatted(protocol, host, projectId, SENTRY_VERSION, SENTRY_CLIENT, publicKey);
+            storeUrl = "%s://%s/api/%s/store/?sentry_version=%s&sentry_client=%s&sentry_key=%s"
+                    .formatted(protocol, host, projectId, SENTRY_VERSION, SENTRY_CLIENT, publicKey);
+
+            envelopeUrl = "%s://%s/api/%s/envelope/?sentry_version=%s&sentry_client=%s&sentry_key=%s"
+                    .formatted(protocol, host, projectId, SENTRY_VERSION, SENTRY_CLIENT, publicKey);
         }
 
-        try (DefaultHttpClient client = new DefaultHttpClient(URI.create(url))) {
+        try (DefaultHttpClient client = new DefaultHttpClient()) {
             String payload = this.payload != null ? runContext.render(this.payload) : runContext.render(DEFAULT_PAYLOAD.strip());
 
-            runContext.logger().debug("Sent the following Sentry event: {}", payload);
+            // Constructing the envelope
+            var eventId = (String) runContext.getVariables().get("eventId");
+            String envelope = constructEnvelope(eventId, payload);
 
-            client.toBlocking().retrieve(HttpRequest.POST(url, payload));
+            // Check envelope and payload against threshold sizes
+            handleThresholds(envelope, payload);
+
+            // Trying to send to /envelope endpoint
+            try {
+                runContext.logger().debug("Attempting to send the following Sentry event envelope: {}", envelope);
+                client.toBlocking().retrieve(HttpRequest.POST(URI.create(envelopeUrl), envelope));
+            } catch (HttpClientResponseException e) { // Backward Compatibility cases
```

Comment:
I'm not a fan of switching from one endpoint to the other if the first didn't work.
Can you tell me more when such event can occurs?

If, for some use cases, the envelope endpoint would never work, maybe it's better to be able to configure which endpoint we want to use and log a friendly error message if the endpoint currently setuped didn't work?

Reply:
the reason to use the old url in catch case was based on the fact of Backward Compatibility from their documentation:

https://develop.sentry.dev/sdk/envelopes/

```
Backward Compatibility
Envelopes require Relay, which has been introduced in Sentry v20.6.0. Earlier versions of Sentry do not support Envelopes and respond with HTTP error 404 Not Found to envelope uploads. Likewise, Relay requires support for Envelopes on the upstream and cannot be used with older versions of Sentry.
```

```
Backward Compatibility
Envelope header authentication requires Relay v21.6.0. Earlier versions of Relay do not support Envelope header authentication and respond with HTTP error 401 Unauthorized ("missing authorization information") to envelope uploads.

SDKs should not rely on Envelope header authentication to retain backward compatibility with older versions of Sentry on-premise unless absolutely required. Instead, stick to HTTP headers or query parameters wherever possible.
````

If we sure that not going with this cases, simply just can keep the single url request with envelop url and in catch providing the error message: like current version isn't support the old '/store endpoint or something like that.

Reply:
OK, so as I thought either of the endpoints will work for a given Sentry account.
So I'll prefer to have a boolean `useEnvelope` (naming for example) or an enum `endpointType` which defaults to envelope but allows to be configured for the old store endpoint.
And in case of 404 or 401 and envelope is configured, log an error (and fail the task of course) with a message explaining that the user may try to configure the store endpoint instead.

Reply:
Maybe then make sense just to support both ?

```
                errors:
                  - id: alert_on_failure
                    type: io.kestra.plugin.notifications.sentry.SentryAlert
                    dsn: "{{ secret('SENTRY_DSN') }}" # format: https://xxx@xxx.ingest.sentry.io/xxx"""
                    endpoint: "store" or envelope ?
```
then in backend side handle if store use old logic if envelope new logic. The enum ov EnvelopeType can have internal logic and let's say generate the url based on the endpoint params.

```
package io.kestra.plugin.notifications.sentry;

public enum EnvelopeType {

    STORE {
        public String buildEnvelopeUrl(String dsn) {
//            return "store";
            String protocol = getProtocol(dsn);
            String publicKey = EnvelopeType.getPublicKey(dsn, protocol);
            String host = getHost(dsn);
            String projectId = EnvelopeType.getProjectId(dsn);
            return "%s://%s/api/%s/store/?sentry_version=%s&sentry_client=%s&sentry_key=%s".formatted(protocol, host, projectId, SENTRY_VERSION, SENTRY_CLIENT, publicKey);
        }
    },
    ENVELOP {
        public String buildEnvelopeUrl(String dsn) {
//            return "envelope";
            String protocol = getProtocol(dsn);
            String publicKey = EnvelopeType.getPublicKey(dsn, protocol);
            String host = getHost(dsn);
            String projectId = EnvelopeType.getProjectId(dsn);
            
            return "%s://%s/api/%s/envelope/?sentry_version=%s&sentry_client=%s&sentry_key=%s".formatted(protocol, host, projectId, SENTRY_VERSION, SENTRY_CLIENT, publicKey);
        }
    };

    public static final String SENTRY_VERSION = "7";
    public static final String SENTRY_CLIENT = "java";
    public static final String SENTRY_KEY = "key";
    
    public abstract String buildEnvelopeUrl(String dsn);

    private static String getProtocol(String dsn) {
        return dsn.split("://")[0];
    }

    private static String getHost(String dsn) {
        return dsn.split("@")[1].split("/")[0];
    }

    private static String getProjectId(String dsn) {
        return dsn.split("@")[1].split("/")[1];
    }

    private static String getPublicKey(String dsn, String protocol) {
        return dsn.split("@")[0].replace(protocol + "://", "");
    }
}
```
if we anyway should support both endpoints as I understood all configurations came from the UI, which in this case the task details.

Reply:
@loicmathieu Shortly said [this commit](https://github.com/kestra-io/plugin-notifications/pull/153/commits/550bbafbb586e2ebcfe40ad82dad842dcbaef0b7)